app/controllers/chapters_controller.rb

```diff
@@ -687,6 +688,7 @@ def chapter_5
     ].to_json
 
     column_chart_data_2 = CsvParser.country_perc('Figure_13.csv', 'Count of PAME evaluations')
+    column_chart_data_2.each{|key,value| column_chart_data_2[key] = commaify(value.to_i)}
```

Comment:
I think the only comment I have is related to having data being formatted in the Controller and using Helpers.
Ideally, helpers should only be used to shape content in the views. Otherwise, data formatting tasks should be performed within a Model, or even better using serialisers.

A Serialiser is essentially a class (bounded to an Entity/Model when possible) which is only responsible of shaping data to then be used in the views. This is mainly to move this kind of logic away from controllers, which could otherwise become a bit messy.

I guess this is an edge case as we dont' have models/database and data is hardcoded.
Also, every other kind of data has already followed the same process anyway, so that would be a significant change.

However, I would strongly suggest to separate the data formatting logic from the controllers in the near future if it can't happen now.

Hope it makes sense and please do let me know your thoughts.

Reply:
Thanks for the comment, Ferdi - this is good to know. I think there is pressure to get the maps added, so this reformatting would probably wait.

If I understand correctly, then the ideal situation would be that the csv in imported to a model and then a serialiser is used to format the data before referencing it in the controller? I.e. the highlighted line here would be in a serialiser.

Is there anything you think can/should be done now without changing the current process for all the data?

Reply:
Yeah I understand the situation.

So, ideally, data should live in the database, then gets fetched in the model and showed by the controller/views.
In this case, we are dealing with static data without importing it into the db. This is still an OK approach in a way, but I think that the data formatting could still go into a Serialiser regardless.
So, even if there are no model specified, you could create a serialiser that gets the raw data as input and produces the desired formatting before sending it into the view.

One way to do this would be to create a `serializers` folder within the `app` one and add serialisers in there for each type of data.

Hope it makes sense.

Reply:
Ah ok, so the serializer doesn't have to be bound to a model. I'll give this ago and let you know when it's done. Thank you for the advice!

Reply:
I've looked into this a bit and am not sure how I would go about using a serializer to take a hash, format it and assign that to an instance variable. Do you have an example of this somewhere? Thanks!